Tests/GRPCNIOTransportHTTP2Tests/HTTP2TransportTLSEnabledTests.swift

```diff
@@ -0,0 +1,444 @@
+/*
+ * Copyright 2024, gRPC Authors All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import Crypto
+import Foundation
+import GRPCNIOTransportHTTP2Posix
+import NIOSSL
+import SwiftASN1
+import Testing
+import X509
+
+@Suite("HTTP/2 transport E2E tests with TLS enabled")
+struct HTTP2TransportTLSEnabledTests {
+  // - MARK: Tests
+
+  @Test(
+    "When using defaults, server does not perform client verification",
+    arguments: [TransportSecurity.posix],
+    [TransportSecurity.posix]
+  )
+  func testRPC_Defaults_OK(
+    clientTransport: TransportSecurity,
+    serverTransport: TransportSecurity
+  ) async throws {
+    let certificateKeyPairs = try SelfSignedCertificateKeyPairs()
+    let clientTransportConfig = self.makeDefaultClientTLSConfig(
+      for: clientTransport,
+      certificateKeyPairs: certificateKeyPairs
+    )
+    let serverTransportConfig = self.makeDefaultServerTLSConfig(
+      for: serverTransport,
+      certificateKeyPairs: certificateKeyPairs
+    )
+
+    try await self.withClientAndServer(
+      clientTransportSecurity: clientTransportConfig,
+      serverTransportSecurity: serverTransportConfig
+    ) { control in
+      await #expect(
+        throws: Never.self,
+        performing: {
+          try await self.executeUnaryRPC(control: control)
+        }
+      )
```

Comment:
nit: prefer single-lines where possible and trailing-closure syntax (same comment applies to other places too):

```suggestion
      await #expect(throws: Never.self) {
        try await self.executeUnaryRPC(control: control)
      }
```